src/index.js

```diff
@@ -1,35 +1,335 @@
 /**
  * @license
- * Copyright 2022 Google LLC
+ * Copyright 2022 MIT
  * SPDX-License-Identifier: Apache-2.0
  */
 
-// TODO: Edit plugin overview.
 /**
- * @fileoverview Plugin overview.
+ * @fileoverview Plugin main class.
  */
 
-// TODO: Rename plugin and edit plugin description.
+import * as Blockly from 'blockly/core';
+
+import DragSelect from '../lib/ds.min';
+
+import * as ContextMenu from './contextmenu';
+import * as Shortcut from './shortcut';
+import {blockSelection, inMultipleSelectionMode, setSelectionMode,
+  origHandleWsStart, hasSelectedParent} from './global';
+import {MultiSelectControls} from './switch';
+
 /**
- * Plugin description.
+ * Class for using multiple select blocks on workspace.
  */
-export class Plugin {
+export class WorkspaceMultiSelect {
   /**
-   * Constructor for ...
-   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the plugin will
-   *     be added to.
+   * Initalize the class data structure.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
    */
   constructor(workspace) {
-    /**
-     * The workspace.
-     * @type {!Blockly.WorkspaceSvg}
-     * @protected
-     */
     this.workspace_ = workspace;
+    this.hasDisableWorkspaceDrag_ = false;
+    this.justUnselectedBlock_ = null;
+    this.blocklyUnselectedBlock_ = null;
+  }
+
+  /**
+   * Bind the events and replace registration.
+   * @param {{multiSelectIcon: boolean, useDoubleClick: boolean}} options
+   * to set.
+   */
+  init(options) {
+    this.onKeyDownWrapper_ = Blockly.browserEvents.conditionalBind(
+        this.workspace_.injectionDiv_, 'keydown', this, this.onKeyDown_);
+    this.onKeyUpWrapper_ = Blockly.browserEvents.conditionalBind(
+        this.workspace_.injectionDiv_, 'keyup', this, this.onKeyUp_);
+    this.eventListenerWrapper_ = this.eventListener_.bind(this);
+    this.workspace_.addChangeListener(this.eventListenerWrapper_);
+
+    ContextMenu.unregisterContextMenu();
+    ContextMenu.registerOurContextMenu();
+    Shortcut.unregisterShortcut();
+    Shortcut.registerOurShortcut();
+
+    if (options.multiSelectIcon) {
+      this.controls_ = new MultiSelectControls(this.workspace_, this);
+      const svgControls = this.controls_.createDom();
+      this.workspace_.svgGroup_.appendChild(svgControls);
+      this.controls_.init();
```

Comment:
You could just pass the config options through here.

Reply:
I passed the config through the constructor as this should be done before Dom is created.